### ISSUE TITLE
NPM Scripts Do Not Handle Special Characters

### ISSUE THREAD

**Issue by Marshal27** [microsoft/vscode#112187]
- VSCode Version: 1.51.1
- OS Version: Windows_NT x64 10.0.19041

Steps to Reproduce:

1. Create sample scripts
```
"test": "npm run '  |_ start (test)'",
"  |_ start (test)": "npm run pre-build && ng serve",
```
2. Observe that running script `test` will work as expected, as the child script name is encapsulated in `'` and thus escaped.
3. Observe that running script `  |_ start (test)` fails with the following exception.
`/bin/bash: -c: line 0: syntax error near unexpected token `('`

Does this issue occur when all extensions are disabled?: Yes

> Both scripts work in IntelliJ without issue, but fail in VSCode due to special characters not being escaped when handed off to `npm run`. 
> 
> Also noticed that the following extension runs the script without issue.
> 
> - https://github.com/Microsoft/vscode-npm-scripts
> 
> Made a simple modification to this extension and recompiled to replace the default NPM SCRIPTS on the left nav bar... I would prefer to not have to modify an external extension to solve this.
> 
> - https://github.com/Duroktar/vscode-npm-scripts
>
> this
> `const command: string = `${packageManager} run ${task}`;`
>
> changed to this
> `const command: string = `${packageManager} run "${task}"`;`

-----------------------------------
**After Looking at This Further**

It appears to be a challenge with task vs handing script off to the terminal. I have narrowed it down to this line, but not sure what the proper combination is to make the task handler happy.

https://github.com/microsoft/vscode/blob/b3d95fea2ab0bf84dc9d23ee6ecebd30131812e2/extensions/npm/src/tasks.ts#L279

**Comment by xmedeko** [comment]
It's probably fixed since 1.53.0

**Comment by Marshal27** [comment]
> It's probably fixed since 1.53.0

confirmed, this is now working in v1.53.0

**Comment by egamma** [comment]
@Marshal27 thanks for the update - closing.
